### ISSUE TITLE
Is it possible to stop the ffmpeg terminal window from opening so much?

### ISSUE THREAD

**Issue by SammyJammy32** [Zulko/moviepy#1908]
Hello!

I'm using moviepy with a lot of audio files and video files and while my script is running, a black ffmpeg terminal window will open and close very fast and repeatedly making it very hard to use my PC while my script is running. Is there any way to prevent this window from opening?

**Comment by koutoftimer** [comment]
@SammyJammy32 What is your OS?

**Comment by SammyJammy32** [comment]
@koutoftimer Windows 10

**Comment by koutoftimer** [comment]
You can check out more general answer related to windows on SO: https://stackoverflow.com/questions/40083575/prevent-pydub-from-opening-console-window

I can't see any possibility to do this (maybe I just missing something) but you can monkey-patch [`__init__`](https://github.com/Zulko/moviepy/blob/master/moviepy/video/io/ffmpeg_writer.py#L140) according to suggestion from SO. Also there is new function callled [`cross_platform_popen_params`](https://github.com/Zulko/moviepy/blob/858bb81fba7e09f1f562283ed6d1394db883b6c7/moviepy/tools.py#L12) that is, probably what you really want to monkey-patch but it is only in the master branch AFAIK (at least I'm using few weeks old release and it is not present yet).

**Comment by SammyJammy32** [comment]
@koutoftimer I appreciate your time and effort with helping me! I am not quite sure what it means to monkey-patch, does that mean going into the library and editing the Popen myself?

**Comment by koutoftimer** [comment]
Something like

```python
import subprocess
from functools import wraps

# save original function
__old_Popen = subprocess.Popen


# create wrapper to be called instead of original one
@wraps(__old_Popen)
def new_Popen(*args, startupinfo=None, **kwargs):

    # way 1, as SO suggests:
    # create window
    startupinfo.dwFlags |= subprocess.STARTF_USESHOWWINDOW
    # and hide it immediately
    startupinfo.wShowWindow = subprocess.SW_HIDE

    # way 2, I cann't test it but you may try just:
    # startupinfo.dwFlags = subprocess.CREATE_NO_WINDOW

    return __old_Popen(*args, startupinfo=startupinfo, **kwargs)


# monkey-patch/replace Popen
subprocess.Popen = new_Popen
```

**Comment by SammyJammy32** [comment]
`File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\moviepy\audio\io\AudioFileClip.py", line 70, in __init__
    self.reader = FFMPEG_AudioReader(filename, fps=fps, nbytes=nbytes,
  File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\moviepy\audio\io\readers.py", line 51, in __init__
    infos = ffmpeg_parse_infos(filename)
  File "C:\Users\User\AppData\Local\Programs\Python\Python310\lib\site-packages\moviepy\video\io\ffmpeg_reader.py", line 257, in ffmpeg_parse_infos
    proc = sp.Popen(cmd, **popen_params)
  File "MyScript.py", line 78, in new_Popen
    startupinfo.dwFlags |= subprocess.STARTF_USESHOWWINDOW
AttributeError: 'NoneType' object has no attribute 'dwFlags'`

@koutoftimer I noticed all the subprocess in moviepy is imported "import subprocess as sp" so I changed all the "subprocess" in what you sent me as "sp". Not sure if I had to but I did just in case. Anyways, I got this error while running it, same thing happens when using that other CREATE_NO_WINDOW line instead

**Comment by koutoftimer** [comment]
Added `None` check.

Python imports modules only once. Any consecutive imports just uses cached version. This is why this monkey-patch is possible, because you are modifying an object that is used by `moviepy`. Will you refer it `sp` or `subprocess` doesn't matter.

**Comment by SammyJammy32** [comment]
Works perfectly! Th

**Comment by koutoftimer** [comment]
@SammyJammy32 issue may be reproduced by others in the future. Do no close an issue until it is resolved in the upstream. If workaround works for you, you may test both version and provide report. This will help anyone who would like to fix this issue.

**Comment by OsaAjani** [comment]
Thank you for your contributions and for reporting issues in this repository. With the release of v2, which introduces significant changes to the codebase and API, we’ve reviewed the backlog of open PRs and issues. Due to the length of the backlog and the likelihood that many of these are either fixed or no longer applicable, we’ve made the decision to close all previous PRs and issues.

If you believe that any of these are still relevant to the current version or if you'd like to reopen a related discussion, please feel free to create a new issue or pull request, referencing the old one.

Thank you for your understanding and continued support!
